### PR TITLE
Pipeline design to support multi scan processing

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: elements_imaging/imaging.py [hunk 1]
```diff
@@ -421,13 +464,18 @@ def make(self, key):
...
 
             # -- summary images --
-            field_keys = (scan.ScanInfo.Field & key).fetch('KEY', order_by='field_z')
+            field_keys = {}
+            for field_key in (Processing.Field * Processing.MemberField
+                              * scan.ScanInfo.Field & key).fetch('KEY', order_by='field_z'):
+                if field_key['processing_field_idx'] not in field_keys:
+                    field_keys[field_key['processing_field_idx']] = field_key
+
             summary_imgs = [{**key, **fkey, 'ref_image': ref_image,
```
Comment (dimitri-yatsenko) [inline elements_imaging/imaging.py:473]: use full words: `summary_images`.  Abbreviations like these do not make the code more readable.

File: elements_imaging/imaging.py [hunk 2]
```diff
@@ -647,6 +695,14 @@ class Trace(dj.Part):
         activity_trace: longblob  # 
         """
 
+    @property
+    def key_source(self):
+        suite2p_ks = (Fluorescence * ActivityExtractionMethod * ProcessingParamSet.proj('processing_method')
```
Comment (dimitri-yatsenko) [inline elements_imaging/imaging.py:700]: avoid abbreviations as they make code less understandable.  Use `suite2p_key_source`

File: elements_imaging/imaging.py [hunk 3]
```diff
@@ -340,7 +340,7 @@ def make(self, key):
 
             # ---- iterate through all s2p plane outputs ----
             rigid_mc, nonrigid_mc, nonrigid_blocks = {}, {}, {}
-            summary_imgs = []
+            summary_images = []
```
Comment (dimitri-yatsenko) [inline elements_imaging/imaging.py:343]: could you please expand the other abbreviations too: `rigid_mc`, `mc_key`, `align_chn`, `ops`.
Also, keep the lines to a reasonable length; I think 95 characters was the length we agreed on.

File: Background.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+We do not include Miniscopes in these estimates. 
+In all there are perhaps on the order of 3000 two-photon setups globally but their processing needs may need to be further segmented.
+
+### Software
++ ScanImage
++ ThorImageLS
++ ScanBox
+
+Vidrio’s ScanImage is the data acquisition software for two types of home-built scanning two-photon systems:
```
Comment (dimitri-yatsenko) [inline Background.md:25]:
```suggestion
Vidrio’s [ScanImage](https://docs.scanimage.org/) is the data acquisition software for two types of home-built scanning two-photon systems: 
```

File: Background.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+In all there are perhaps on the order of 3000 two-photon setups globally but their processing needs may need to be further segmented.
+
+### Software
++ ScanImage
++ ThorImageLS
++ ScanBox
+
+Vidrio’s ScanImage is the data acquisition software for two types of home-built scanning two-photon systems: 
+based on Thorlabs and Sutter hardware. ScanImage has a free version and a licensed version. 
+They have about 200 active paid licenses.
```
Comment (dimitri-yatsenko) [inline Background.md:27]: no need to state their numbers publicly - not confirmed anyway.

**Review by dimitri-yatsenko — APPROVED** [review]

**Review by shenshan — APPROVED** [review]

File: element_calcium_imaging/scan.py [hunk 6]
```diff
@@ -243,14 +246,17 @@ def make(self, key):
                                         field_idx=plane_idx,
                                         px_height=px_height,
                                         px_width=px_width,
-                                        um_height=px_height * sbx_meta['um_per_pixel_y'] if sbx_meta['um_per_pixel_y'] else None,
-                                        um_width=px_width * sbx_meta['um_per_pixel_x'] if sbx_meta['um_per_pixel_x'] else None,
+                                        um_height=px_height * sbx_meta['um_per_pixel_y']
+                                        if sbx_meta['um_per_pixel_y'] else None,
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/scan.py:250]: weird formatting

**Review by dimitri-yatsenko — APPROVED** [review]

**Review by dimitri-yatsenko — APPROVED** [review]

**Pull request merged**